### PR TITLE
Listings

### LINKED ISSUE #19: Backend can query listings
- [ ] return all listings that are less than 72 hours old 
- [ ] group listings by produce_name 
- [ ] return grouped listings by alphabetical produce_name 
- [ ] listing data to display: type, quantity, unit, zip

### PR DESCRIPTION
**Pull request by jdcarey128**
## Description

This PR defines the getListings and showListing query. The getListings returns JSON data where listings are grouped by their produce_name and returned as arrays within those groupings. The showListing returns the details for a specific listing given the id as an argument. 

This PR also organizes the spec files into a queries and mutations subdirectory for better organization. 

UPDATE 
This PR also adds the getUserListings query 

Closes #19, #17, #10

## Motivation and Context

The getListings query is more complicated so that the front end doesn't need to do much filtering to organize the listings into their respective cards. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/graphql/queries/get_listings.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    field :error, [String], null: false 
+
+    def resolve
+      error = []
+      three_days_ago = DateTime.now - 3.days
+      listings = Listing.where('date_harvested >= ?', three_days_ago).order('produce_name').group_by(&:produce_name)
+      error << 'There are no recent produce listings' if listings.empty?
+      {
+        listings: listings, 
+        error: error
```
Comment (Callbritton) [inline app/graphql/queries/get_listings.rb:15]: I will be sure to add this to the produce queries!

**Review by Callbritton — APPROVED** [review]

**Review by dcoleman21 — APPROVED** [review]
Looks like you added a lot of functionality here.

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -6,5 +6,6 @@ class QueryType < Types::BaseObject
 
     field :show_listing, resolver: Queries::ShowListing
     field :get_listings, resolver: Queries::GetListings
```
Comment (dcoleman21) [inline app/graphql/types/query_type.rb:8]: So this query get_listings is getting all the listings?

File: app/graphql/types/query_type.rb [hunk 3]
```diff
@@ -6,5 +6,6 @@ class QueryType < Types::BaseObject
...
+    field :get_user_listings, resolver: Queries::GetUserListings
```
Comment (dcoleman21) [inline app/graphql/types/query_type.rb:9]: And this query is getting listing by user_id?

**Review by dcoleman21 — APPROVED** [review]
I see that you had to differentiate between getListing and getUserListing. I look forward to gathering as a group to discuss our progress.

**Pull request merged**